internal/host/host.go

```diff
@@ -548,14 +548,19 @@ func (m *Manager) UpdateInstallProgress(ctx context.Context, h *models.Host, pro
 			"progress_progress_info":    progress.ProgressInfo,
 			"progress_stage_updated_at": strfmt.DateTime(time.Now()),
 		}
-		return m.db.Model(h).UpdateColumns(updates).Error
+		if err := m.db.Model(h).UpdateColumns(updates).Error; err != nil {
+			return common.NewApiError(http.StatusInternalServerError, err)
```

Comment:
I'd say just return the error here because we map the gorm not found errors to 404 in `GenerateErrorResponder`.

Reply:
Gotcha, modified as such! Thanks!